src/Cake.Common/Tools/DNX/Run/DNXRunSettings.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.DNX.Run
+{
+    /// <summary>
+    /// Contains settings used by <see cref="DNXRunner"/>
+    /// </summary>
+    public class DNXRunSettings : ToolSettings
+    {
+        /// <summary>
+        /// Gets or sets the project dir to be used (see dnx --project option)
+        /// </summary>
+        public string Project { get; set; }
+
+        /// <summary>
+        /// Gets or sets the framework to be used when running dnx
+        /// </summary>
+        public string Framework { get; set; }
+
+        /// <summary>
+        /// Gets or sets the configuration to be used when running dnx (see dnx --configuration option)
+        /// </summary>
+        public string Configuration { get; set; }
+
+        /// <summary>
+        /// Gets or sets the app base dir to be used by dnx (see dnx --appbase option)
+        /// </summary>
+        public string AppBase { get; set; }
+
+        /// <summary>
+        /// Gets or sets the lib dir to be used by dnx (see dnx --lib option)
+        /// </summary>
+        public string Lib { get; set; }

```

Comment:
Actually I would prefer an ICollection here for the sake of consistency in
the API.

It's not a bad idea to change this, but it would be better if we changed
all APIs that use this at the same time (in the future) so it will remain
consistent.
On Jan 6, 2016 11:52, "Vincent Lainé" <notifications@github.com> wrote:

> In src/Cake.Common/Tools/DNX/Run/DNXRunSettings.cs
> <https://github.com/cake-build/cake/pull/591#discussion_r48945960>:
>
> > +        public string Framework { get; set; }
> > +
> > +        /// <summary>
> > +        /// Gets or sets the configuration to be used when running dnx (see dnx --configuration option)
> > +        /// </summary>
> > +        public string Configuration { get; set; }
> > +
> > +        /// <summary>
> > +        /// Gets or sets the app base dir to be used by dnx (see dnx --appbase option)
> > +        /// </summary>
> > +        public string AppBase { get; set; }
> > +
> > +        /// <summary>
> > +        /// Gets or sets the lib dir to be used by dnx (see dnx --lib option)
> > +        /// </summary>
> > +        public string Lib { get; set; }
>
> I used a DirectoryPathCollection instead of ICollection with the default
> PathComparer.
> this.Lib = new DirectoryPathCollection(PathComparer.Default);
>
> Is ok for you ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cake-build/cake/pull/591/files#r48945960>.
>